src/unix.rs

```diff
@@ -219,20 +281,54 @@ pub fn thread_schedule_policy() -> Result<ThreadSchedulePolicy, Error> {
 /// let thread_id = thread_native_id();
 /// let policy = ThreadSchedulePolicy::Realtime(RealtimeThreadSchedulePolicy::Fifo);
 /// let params = ScheduleParams { sched_priority: 3 as libc::c_int };
-/// assert!(set_thread_schedule_policy(thread_id, policy, params).is_ok());
+/// let priority = ThreadPriority::Min;
+/// assert!(set_thread_schedule_policy(thread_id, policy, params, priority).is_ok());
 /// ```
 pub fn set_thread_schedule_policy(
     native: ThreadId,
     policy: ThreadSchedulePolicy,
     params: ScheduleParams,
+    priority: ThreadPriority,
 ) -> Result<(), Error> {
     let params = params.into_posix();
     unsafe {
-        let ret = libc::pthread_setschedparam(
-            native,
-            policy.to_posix(),
-            &params as *const libc::sched_param,
-        );
+        let ret = match policy {
+            // SCHED_DEADLINE policy requires its own syscall
+            #[cfg(target_os = "linux")]
+            ThreadSchedulePolicy::Realtime(RealtimeThreadSchedulePolicy::Deadline) => {
+                let (runtime, deadline, period) = match priority {
+                    ThreadPriority::Deadline(r, d, p) => (r, d, p),
+                    _ => {
+                        return Err(Error::Priority(
+                            "Deadline policy given without deadline priority.",
+                        ))
+                    }
+                };
+                let tid = native as libc::pid_t;
+                let sched_attr = SchedAttr {
+                    size: std::mem::size_of::<SchedAttr>() as u32,
+                    sched_policy: policy.to_posix() as u32,
+
+                    sched_runtime: runtime as u64,
+                    sched_deadline: deadline as u64,
+                    sched_period: period as u64,
+
+                    ..Default::default()
+                };
+                libc::syscall(
+                    libc::SYS_sched_setattr,
+                    tid,
+                    &sched_attr as *const _,
+                    // we are not setting SCHED_FLAG_RECLAIM nor SCHED_FLAG_DL_OVERRUN
```

Comment:
Just curious, why don't we set it? Is it possible to set it?

Reply:
I am not using such a policy anywhere, do you think it is worth adding these options to the interface so that others could configure them?

Reply:
Sure! Thanks.